evaluation/scripts/wmt.py

```diff
@@ -0,0 +1,52 @@
+import argparse
+
+import torch
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+from ..datasets.wmt import WMTEnglishDataset
+
+
+def main(args):
+    """adapted from https://huggingface.co/transformers/perplexity.html"""
+    stride = args.stride
+    device = torch.device(args.device)
+    tokenizer = AutoTokenizer.from_pretrained(args.model)
+    model = AutoModelForCausalLM.from_pretrained(args.model).to(device)
+    dataset = WMTEnglishDataset(
+        tokenizer,
+        stride=stride,
+        max_len=model.config.n_positions,
+        pair=args.pair,
+    )
+    loader = DataLoader(
+        dataset,
+        batch_size=args.batch_size,
+        num_workers=args.num_workers,
+        shuffle=False,
+        drop_last=True,
+    )
+    log_likelihoods = []
+    for input_ids in tqdm(loader):
+        input_ids = input_ids.to(device)
+        target_ids = input_ids.clone()
+        target_ids[:, :-stride] = -100
+        with torch.no_grad():
+            outputs = model(input_ids, labels=target_ids)
+            log_likelihood = outputs[0]
```

Comment:
This is a minor point. I prefer to compute the total log_likelihood of a batch to improve code readability. Need to compute a `target_length` before this line.

```suggestion
            log_likelihood = outputs[0] * target_length
```